### PR TITLE
[REVIEW] Pin `dask` and `distributed` for release

### PR DESCRIPTION
**Pull request by galipremsagar**
This PR pins `dask` and `distributed` to `2023.3.2` and `2023.3.2.1` respectively for `23.04` release.

xref: https://github.com/rapidsai/cudf/pull/13070

### CODE REVIEW

**Review by pentschev — APPROVED** [review]
LGTM, thanks @galipremsagar !

**Review by raydouglass — APPROVED** [review]

File: dependencies.yaml [hunk 1]
```diff
@@ -95,8 +95,8 @@ dependencies:
     common:
       - output_types: [conda, requirements]
         packages:
-          - dask>=2023.1.1
-          - distributed>=2023.1.1
+          - dask==2023.3.2
+          - distributed==2023.3.2.1
```
Comment (galipremsagar) [inline dependencies.yaml:99]:
```suggestion
          - dask==2023.3.2
          - dask-core==2023.3.2
          - distributed==2023.3.2.1
```

File: pyproject.toml [hunk 2]
```diff
@@ -19,8 +19,8 @@ authors = [
 license = { text = "Apache-2.0" }
 requires-python = ">=3.8"
 dependencies = [
-    "dask >=2023.1.1",
-    "distributed >=2023.1.1",
+    "dask ==2023.3.2",
+    "distributed ==2023.3.2.1",
```
Comment (galipremsagar) [inline pyproject.toml:23]:
```suggestion
    "dask ==2023.3.2",
    "dask-core ==2023.3.2",
    "distributed ==2023.3.2.1",
```

**Comment by quasiben** [comment]
Woot!

**Pull request merged**